plugins/org.jboss.tools.openshift.common.core/src/org/jboss/tools/openshift/common/core/connection/AbstractConnection.java

```diff
@@ -87,4 +89,20 @@ public boolean equals(Object obj) {
 		}
 		return true;
 	}
+
+	@Override
+	public boolean credentialsEqual(IConnection connection) {
+		if(!equals(connection)) {
+			return false;
+		}
+		//It is safe to cast now.
+		AbstractConnection other = (AbstractConnection)connection;
+		if(!Objects.equals(getUsername(), other.getUsername())) {
+			return false;
+		}
+		if(!Objects.equals(getPassword(), other.getPassword())) {
+			return false;
+		}
+		return true;
+	}
 }
```

Comment:
Having a special #equals for credentials seems legit since we consider 2 connections as identical if they only differ in authentication means (ex. password or token). On the other hand the username is used to persist them and thus is required to identify and #equal 2 connections. I thus dont think that #equals may be true while Objects.equals(getUsername(), other.getUsername()) returns false.